Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+					lblFeedback.Text = reason;
+				}
+				return;
+			}
+
+
+			this.Canceled = false;
+			Application.RequestStop ();
+		}
+
+		private void NavigateIf (KeyEventEventArgs keyEvent, Key isKey, View to)
+		{
+			if (!keyEvent.Handled) {
+
+				if (NavigateIf (keyEvent.KeyEvent, isKey, to)) {
+					keyEvent.Handled = true;
+				}
+			}
+		}
+
+		private bool NavigateIf (KeyEvent keyEvent, Key isKey, View to)
+		{
+			if (keyEvent.Key == isKey) {
+
+				to.FocusFirst ();
+				if (to == tbPath) {
+					tbPath.MoveCursorToEnd ();
+				}
+				return true;
+			}
+
+			return false;
+		}
+
+		private void TreeView_SelectionChanged (object sender, SelectionChangedEventArgs<object> e)
+		{
+			if (e.NewValue == null) {
+				return;
+			}
+
+			this.tbPath.Text = FileDialogTreeBuilder.NodeToDirectory (e.NewValue).FullName;
+		}
+
+		private bool IsValidSpecialFolder (SpecialFolder arg)
+		{
+			try {
+				var path = Environment.GetFolderPath (arg);
+				return !string.IsNullOrWhiteSpace (path) && Directory.Exists (path);
+			} catch (Exception) {
+
+				return false;
+			}
+		}
+		private FileDialogRootTreeNode GetTreeNode (SpecialFolder arg)
+		{
+			return new FileDialogRootTreeNode (
+				arg.ToString (),
+				new DirectoryInfo (Environment.GetFolderPath (arg)));
+		}
+
+		private void UpdateNavigationVisibility ()
+		{
+			this.lblBack.Visible = this.history.CanBack ();
+			this.lblForward.Visible = this.history.CanForward ();
+			this.lblUp.Visible = this.history.CanUp ();
+		}
+
+		private void TableView_SelectedCellChanged (TableView.SelectedCellChangedEventArgs obj)
+		{
+			if (!this.tableView.HasFocus || obj.NewRow == -1 || obj.Table.Rows.Count == 0) {
+				return;
+			}
+
+			if (this.tableView.MultiSelect && this.tableView.MultiSelectedRegions.Any ()) {
+				return;
+			}
+
+			var stats = this.RowToStats (obj.NewRow);
+
+			if (stats == null) {
+				return;
+			}
+			FileSystemInfo dest;
+
+			if (stats.IsParent) {
+				dest = state.Directory;
+			} else {
+				dest = stats.FileSystemInfo;
+			}
+
+			try {
+				this.pushingState = true;
+
+				this.tbPath.SetTextTo (dest);
+				this.state.Selected = stats;
+				this.tbPath.ClearSuggestions ();
+
+			} finally {
+
+				this.pushingState = false;
+			}
+		}
+
+
+		private void Combo_SelectedItemChanged (ComboBox combo, ListViewItemEventArgs obj)
+		{
+			var allow = combo.Source.ToList () [obj.Item] as AllowedType;
+			this.currentFilter = allow == null || allow.IsAny ? null : allow;
+
+			this.tbPath.ClearAllSelection ();
+			this.tbPath.ClearSuggestions ();
+
+			if (this.state != null) {
+				this.state.RefreshChildren ();
+				this.WriteStateToTableView ();
+			}
+		}
+
+		private bool TableView_KeyUp (KeyEvent keyEvent)
+		{
+			if (keyEvent.Key == Key.Backspace) {
+				return this.history.Back ();
+			}
+			if (keyEvent.Key == (Key.ShiftMask | Key.Backspace)) {
+				return this.history.Forward ();
+			}
+
+			return false;
+		}
+
+		private void SetupColorSchemes ()
```

Comment:
Move these to ConfigMan. I'll update the docs, but best practice is:

- Put the defaults into `Resources/config.json`
- Don't set default colors in code (use `-1`)
- If reasonable, add a check that throws an exception if a Color is `-1` ("Configuration was not loaded properly").